internal/rootcoord/task.go

```diff
@@ -895,3 +895,204 @@ func (t *DropIndexReqTask) Execute(ctx context.Context) error {
+	ts, err := t.core.TSOAllocator(1)
+	if err != nil {
+		return fmt.Errorf("TSO alloc fail, error = %w", err)
+	}
+
+	// use lambda function here to guarantee all resources to be released
+	createAliasFn := func() error {
+		// lock for ddl operation
+		t.core.ddlLock.Lock()
+		defer t.core.ddlLock.Unlock()
+
+		t.core.chanTimeTick.AddDdlTimeTick(ts, reason)
+		// clear ddl timetick in all conditions
+		defer t.core.chanTimeTick.RemoveDdlTimeTick(ts, reason)
+		err = t.core.MetaTable.AddAlias(t.Req.Alias, t.Req.CollectionName, ts, ddOp)
+		if err != nil {
+			return err
+		}
+		t.core.SendTimeTick(ts, reason)
+		return nil
+	}
+
+	err = createAliasFn()
+	if err != nil {
+		return err
+	}
+
+	// Update DDOperation in etcd
+	return t.core.setDdMsgSendFlag(true)
+}
+
+type DropAliasReqTask struct {
+	baseReqTask
+	Req *milvuspb.DropAliasRequest
+}
+
+func (t *DropAliasReqTask) Type() commonpb.MsgType {
+	return t.Req.Base.MsgType
+}
+
+func (t *DropAliasReqTask) Execute(ctx context.Context) error {
+	if t.Type() != commonpb.MsgType_DropAlias {
+		return fmt.Errorf("create alias, msg type = %s", commonpb.MsgType_name[int32(t.Type())])
+	}
+
+	ddReq := internalpb.DropAliasRequest{
+		Base:  t.Req.Base,
+		Alias: t.Req.Alias,
+	}
+
+	// build DdOperation and save it into etcd, when ddmsg send fail,
+	// system can restore ddmsg from etcd and re-send
+	ddOp := func(ts typeutil.Timestamp) (string, error) {
+		ddReq.Base.Timestamp = ts
+		return EncodeDdOperation(&ddReq, DropAliasDDType)
+	}
+
+	reason := fmt.Sprintf("create alias %s", t.Req.Alias)
+	ts, err := t.core.TSOAllocator(1)
+	if err != nil {
+		return fmt.Errorf("TSO alloc fail, error = %w", err)
+	}
+
+	// use lambda function here to guarantee all resources to be released
+	dropAliasFn := func() error {
+		// lock for ddl operation
+		t.core.ddlLock.Lock()
+		defer t.core.ddlLock.Unlock()
+
+		t.core.chanTimeTick.AddDdlTimeTick(ts, reason)
+		// clear ddl timetick in all conditions
+		defer t.core.chanTimeTick.RemoveDdlTimeTick(ts, reason)
+		err = t.core.MetaTable.DeleteAlias(t.Req.Alias, ts, ddOp)
+		if err != nil {
+			return err
+		}
+		t.core.SendTimeTick(ts, reason)
+		return nil
+	}
+
+	err = dropAliasFn()
+	if err != nil {
+		return err
+	}
+
+	req := proxypb.InvalidateCollMetaCacheRequest{
+		Base: &commonpb.MsgBase{
+			MsgType:   0, //TODO, msg type
+			MsgID:     0, //TODO, msg id
+			Timestamp: ts,
+			SourceID:  t.core.session.ServerID,
+		},
+		CollectionName: t.Req.Alias,
+	}
+	// error doesn't matter here
+	t.core.proxyClientManager.InvalidateCollectionMetaCache(ctx, &req)
+
+	// Update DDOperation in etcd
+	return t.core.setDdMsgSendFlag(true)
+
+}
+
+type AlterAliasReqTask struct {
+	baseReqTask
+	Req *milvuspb.AlterAliasRequest
+}
+
+func (t *AlterAliasReqTask) Type() commonpb.MsgType {
+	return t.Req.Base.MsgType
+}
+
+func (t *AlterAliasReqTask) Execute(ctx context.Context) error {
+	if t.Type() != commonpb.MsgType_AlterAlias {
```

Comment:
what's the expect behaviour if user alter alias then access from the old alias?
Do we need to let proxy know and clean alias cache through something like InvalidateCollectionMetaCache?

Reply:
Yes. I've added invalidate cache part in this PR for `DropAlias/AlterAlias`. 
https://github.com/milvus-io/milvus/pull/7255/files#diff-303ea8a50d3afa52fc5ad6e7342ed1df631c1729c34e41766027138fcdbf91c3R1093